tenets.md

```diff
@@ -0,0 +1,428 @@
+are an opportunity for bugs.
+They also work against inter-operability,
+because a unique behavior in one implementation
+may be relied on by customers.
+If this behavior does not exist
+in other implementations,
+then along this dimension
+these implementations are not inter-operable.
+
+For customers this means
+as we design features,
+they will be better understood
+and as they develop over time they should trend
+towards a consistent idea.
+
+## Easy to use/Hard to misuse
+
+### Summary
+
+This principle is often referenced as one
+but is two parts.
+Discussing them individually helps
+clarify what each one means
+and how they reinforce each other.
+
+## Easy to use
+
+### Summary
+
+Tools should be easy to use.
+The form should lend itself to the function.
+The way that these tools work together
+should be obvious.
+
+An example is furniture assembly.
+Pre-drilled holes with nuts and bolts
+make it clear both how and where
+parts should be joined.
+The shape of the joints
+can also resolve left and right
+or front and back.
+
+The AWS Encryption SDK (ESDK) is a primitive that
+is used to build larger software systems.
+It is composed of parts and options.
+
+Customers use these parts together
+to accomplish their goals.
+The ESDK MUST facilitate the customer goal,
+not frustrate it.
+
+### Motivation
+
+Our priority is to
+design parts that intuitively compose together.
+[Design for assembly](https://en.wikipedia.org/wiki/Design_for_assembly)
+emphasizes creating a functioning product from parts.
+
+The ESDK is a part inside our customer's application.
+Our APIs SHOULD explain to our customers
+how they fit together.
+
+The time our customers spend
+trying to figure out how to use the ESDK
+is just a transaction cost
+that takes away from their goal.
+
+## Hard to misuse
+
+### Summary
+
+Tools should be safe.
+It should be clear to customers
+if there are sharp edges and where.
+
+An example of "hard to misuse"
+is gas and water pipes.
+In the US they have opposite threading.
+This makes it very difficult to connect
+a gas line to a water line accidentally.
+
+The AWS Encryption SDK (ESDK) SHOULD be [easy to use](#easy-to-use)
+but SHOULD NOT easily go together unsafely.
+
+Customers should not be surprised
+by unexpected or unsafe behavior.
+
+### Motivation
+
+Not every configuration is safe to use.
+Expressiveness and power
+can be at odds with safety.
+Therefore, we actively discourage unsafe use.
+This is another property you get with [design for assembly](https://en.wikipedia.org/wiki/Design_for_assembly).
+
+This means we will decline to build features
+if the danger of misuse is too high.
+This is not simply a case of cost vs. benefit.
+The downside risk of misuse SHOULD be evaluated
+independent of the value of new features.
+
+## Correct by construction
+
+### Summary
+
+To verify that code is correct
+there are three stages
+in which incorrectness can be identified.
+
+1. Development
```

Comment:
We define these three stages but then do not refer to them again. What point are we trying to drive here?

Reply:
I think I've fixed this. Take a look